### DISCUSSION TITLE
Basketball: Getting player stats per day

### DISCUSSION THREAD

**Discussion by raag16** [cwendt94/espn-api#611] [Q&A]
I am able to get a player's stats for the latest day (11-17) but beyond that can only get averages or totals for the matchup period or season. If for example, I want Tyus Jones' stats for 11-15, how might I be able to do that? My larger goal is to get metrics such as "how many players have played over the course of the h2h matchup for my team and my opponent" and many more more for which I need individual players' daily stats. Thanks!

**Comment by cwendt94** [comment]
I believe this should help with your question https://github.com/cwendt94/espn-api/discussions/594. You can pass in scoring_period for box_scores which is the single day of the matchup. This should return players stats for that day.

**Reply by raag16** [reply]
I can get their 2025_total or last 7 days averages stats (and a few others like this) but using that response, I am unable to get their individual stats for that specific day (e.g. points, rebounds, assists).

**Reply by cwendt94** [reply]
The box score will get their stats for that matchup and can for those individual games. However is you want to get all of a players stats for the whole season you can use `player_info` call and pass in the playerId. This will return a player class and the `stats` field will have all the games they played. It would look like this
```python3
# Tyrese Haliburton
player = league.player_info(playerId=4396993)
```

**Reply by austintedesco** [reply]
@raag16 - Something like the below is working well for me. It returns data in a table format, where each row represents statistics from one NBA game played by one player. It includes all NBA games over the entire season to date. You can modify this to bring in additional columns that might be useful for your purposes, but this should give you the most granular data that you can aggregate as desired.


```
# Initialize League
league = League(league_id=league_id, year=year, espn_s2=espn_s2, swid=swid)

# List of all player IDs
player_ids = list(league.player_map.keys())

# Store extracted data
nba_game_data = []
i = 0

# Loop through every player to retrieve stats
for player_id in player_ids:
    player = league.player_info(playerId=player_id)

    # Skip if player data is missing
    if not player:
        # print(f"Skipping playerId {player_id} (No data found)")
        # i=i+1
        # print(i)
        continue

    player_name = player.name

    for game_id, game_stats in player.stats.items():
        if "total" in game_stats and game_id in player.schedule:
            nba_game_data.append({
                "playerId": player_id,
                "player_name": player_name,
                "game_id": game_id,
                "date": player.schedule[game_id]["date"].strftime("%Y-%m-%d"),  # Format date
                "team": player.schedule[game_id]["team"],  # Opponent team
                "applied_total": game_stats.get("applied_total", 0),  # Fantasy points
                **game_stats["total"]  # Include all raw stats
            })

# Convert to DataFrame
df = pd.DataFrame(nba_game_data)

# Save to Excel
df.to_excel("nba_2025_game_stats.xlsx", index=False)

print("Excel file created: nba_2025_game_stats.xlsx")
```
